working-with-openai/open-ai-api-endpoints.md

```diff
@@ -1,153 +1,191 @@
+### Text Completion Response Body
 
-### Chat Completion Response Body
+After running the above request, we received the response below. Your response may look different, but the properties should be the same or similar. 
 
-After running the above request, we received the response below. Your response may look different, but the properties should be the same or similar. This response is known as a chat completions object.
 ```
 {
-    "id": "chatcmpl-9bBybziJstOHt9FAWJX8MOWA8QJ2B",
-    "object": "chat.completion",
-    "created": 1718651201,
-    "model": "gpt-3.5-turbo-0125",
-    "choices": [
-        {
-            "index": 0,
-            "message": {
-                "role": "assistant",
-                "content": "Yes, I can understand your message. How can I assist you today?"
-            },
-            "logprobs": null,
-            "finish_reason": "stop"
-        }
-    ],
-    "usage": {
-        "prompt_tokens": 26,
-        "completion_tokens": 15,
-        "total_tokens": 41
-    },
-    "system_fingerprint": null
+  "candidates": [
+    {
+      "content": {
+        "parts": [
+          {
+            "text": "Each cat has 4 paws, and you have 3 cats, so there are 3 * 4 = 12 paws in your home.\n"
+          }
+        ],
+        "role": "model"
+      },
+      "finishReason": "STOP",
+      "avgLogprobs": -0.006260416004806757
+    }
+  ],
+  "usageMetadata": {
+    "promptTokenCount": 30,
+    "candidatesTokenCount": 32,
+    "totalTokenCount": 62
+  },
+  "modelVersion": "gemini-1.5-flash"
 }
 ```
 
-| Piece of Code | Notes |
+| <div style="min-width:150px;">Piece of Code</div> | Notes |
 | ------------- | ----- | 
-| `id` | A unique identifier for the response |
-| `model` | The model that was used |
-| `choices` | A list of potential responses that you can choose from. If you did not specify an n in the request body, there will only be one choice. |
-| `usage` | Details on how many tokens were used in the request and response |
+| `candidates` | A list of potential responses that you can choose from. There are changes around this underway but typically there will only be one choice. |
+| `modelVersion` | The model that was used |
+| `usageMetadata` | Details on how many tokens were used in the request and response |
+
+You have now written your first request using a history of messages to the Gemini API! 
+
+To read through the API's generated response to our prompt, we will need to: 
+1. iterate through the `candidates` list 
+2. for each element in `candidates` access the `content` key  
+3. access the `parts` key and iterate over the content's `parts` list 
+4. for each element in `parts` read the `text` response and handle that string however we would like.  
 
-You have now written your first request to the OpenAI API! 
+This might sound a little tedious, but in our upcoming case study we'll show how we can use Google's `google.generativeai` Python library to make accessing the response text a little easier.
```

Comment:
```suggestion
This might sound a little tedious, but in our upcoming case study we'll show how we can use Google's `[google-generativeai](https://pypi.org/project/google-generativeai/)` Python package to make accessing the response text a little easier.
```